samples/java-spring-eventsourced-customer-registry-subscriber/src/it/java/customer/api/CustomerIntegrationTest.java

```diff
@@ -1,92 +1,171 @@
 import org.hamcrest.core.IsEqual;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.junit.jupiter.api.*;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.context.annotation.Import;
+import org.springframework.context.ApplicationContext;
 import org.springframework.http.*;
+import org.springframework.http.codec.json.Jackson2JsonEncoder;
 import org.springframework.web.reactive.function.client.WebClient;
 import org.springframework.web.reactive.function.client.WebClientRequestException;
 import reactor.core.publisher.Mono;
 
 import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
 
 import static java.time.temporal.ChronoUnit.SECONDS;
 import static org.awaitility.Awaitility.await;
 
-
+/**
+ * This test exercises the integtration between the current service (customer-registry-subscriber) and the customer-registry service.
+ * 
+ * The customer registry service is started as a docker container as well as it own kalix proxy. The current service is as a local JVM process (not dockerized),
+ * but its own kalix proxy starts as a docker container. The `docker-compose-integration.yml` file is used to start all these services.
+ * 
+ * The subscriber service will first create a customer on customer-registry service. The customer will be streamed back to the subscriber service and update its view.
+ * 
+ * This test will exercise the following:
+ * - servcice under test can read settings from docker-compose file and correctly configured itself. 
+ * - resolution of service port mappings from docker-compose file allows for cross service calls (eg: create customer from subscriber service)
+ * - resolution of service port mappings passed to kalix-proxy allows for service to service streaming (eg: customer view is updated in subscriber service)
+ */
 @SpringBootTest(classes = Main.class)
-@Import(TestkitConfig.class)
-public class CustomerIntegrationTest extends KalixIntegrationTestKitSupport {
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CustomerIntegrationTest {
+
+  final private Duration timeout = Duration.of(5, SECONDS);
+
+  private DockerComposeUtils dockerComposeUtils = new DockerComposeUtils("docker-compose-integration.yml");
+
+  private KalixSpringApplication kalixSpringApplication;
+
+  public CustomerIntegrationTest(ApplicationContext applicationContext) {
+    Map<String, Object> confMap = new HashMap<>();
+    confMap.put("kalix.user-function-port", dockerComposeUtils.userFunctionPort());
+    // don't kill the test JVM when terminating the KalixRunner
+    confMap.put("kalix.system.akka.coordinated-shutdown.exit-jvm", "off");
+    // dev-mode should be false when running integration tests
+    confMap.put("kalix.dev-mode.enabled", false);
+    confMap.put("kalix.user-function-interface", "0.0.0.0");
```

Comment:
Last but not least, this little guy was a major time consumer.

Not needed on Mac, but required for Linux. The test was 'mysteriously' failing in CI because of this. 🤦

Reply:
Yes, by default is localhost and it fails for Linux.